homeassistant/components/device_tracker/freebox.py

```diff
@@ -0,0 +1,144 @@
+
+REQUIREMENTS = ['aiofreepybox==0.0.3']
+
+_LOGGER = logging.getLogger(__name__)
+
+FREEBOX_CONFIG_FILE = 'freebox.conf'
+
+PLATFORM_SCHEMA = vol.All(
+    PLATFORM_SCHEMA.extend({
+        vol.Required(CONF_HOST): cv.string,
+        vol.Required(CONF_PORT): cv.port
+    }))
+
+MIN_TIME_BETWEEN_SCANS = timedelta(seconds=10)
+
+
+async def async_setup_scanner(hass, config, async_see, discovery_info=None):
+    """Set up the Freebox device tracker and start the polling."""
+    freebox_config = copy.deepcopy(config)
+    if discovery_info is not None:
+        freebox_config[CONF_HOST] = discovery_info['properties']['api_domain']
+        freebox_config[CONF_PORT] = discovery_info['properties']['https_port']
+        _LOGGER.info("Discovered Freebox server: %s:%s",
+                     freebox_config[CONF_HOST], freebox_config[CONF_PORT])
+
+    scanner = FreeboxDeviceScanner(hass, freebox_config, async_see)
+    interval = freebox_config.get(CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL)
+    await scanner.async_start(hass, interval)
+    return True
+
+
+Device = namedtuple('Device', ['id', 'name', 'ip'])
+
+
+def _build_device(device_dict):
+    return Device(
+        device_dict['l2ident']['id'],
+        device_dict['primary_name'],
+        device_dict['l3connectivities'][0]['addr'])
+
+
+class FreeboxDeviceScanner(object):
+    """This class scans for devices connected to the Freebox."""
+
+    def __init__(self, hass, config, async_see):
+        """Initialize the scanner."""
+        from aiofreepybox import Freepybox
+        import asyncio
+        import socket
+
+        self.host = config[CONF_HOST]
+        self.port = config[CONF_PORT]
+        self.token_file = hass.config.path(FREEBOX_CONFIG_FILE)
+        self.async_see = async_see
+
+        # Hardcode the app description to avoid invalidating the authentication
+        # file at each new version.
+        # The version can be changed if we want the user to re-authorize HASS
+        # on her Freebox.
+        app_desc = {
+            'app_id': 'hass',
+            'app_name': 'Home Assistant',
+            'app_version': '0.65',
+            'device_name': socket.gethostname()
+        }
+
+        api_version = 'v1'  # Use the lowest working version.
+        self.fbx = Freepybox(
+            app_desc=app_desc,
+            token_file=self.token_file,
+            api_version=api_version)
+
+    async def async_start(self, hass, interval):
+        """Perform a first update and start pooling at the given interval."""
+        await self.async_update_info()
+        async_track_time_interval(hass, self.async_update_info, interval)
+
+    @Throttle(MIN_TIME_BETWEEN_SCANS)
+    async def async_update_info(self, now=None):
+        """Check the Freebox for devices."""
+        _LOGGER.info('Scanning devices')
+
+        import asyncio
+
+        await self.fbx.open(self.host, self.port)
+        try:
+            hosts = await self.fbx.lan.get_hosts_list()
+        finally:
```

Comment:
If we expect an exception here, maybe we should try to catch it, if we know what it is and if it will happen often?